lightning/src/routing/onion_message.rs

```diff
@@ -0,0 +1,280 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! [Onion message] pathfinding lives here.
+//!
+//! Finding paths for onion messages is necessary for retrieving invoices and fulfilling invoice
+//! requests from [offers]. It differs from payment pathfinding in that channel liquidity, fees, and
+//! knobs such as `htlc_maximum_msat` do not factor into path selection -- onion messages require a
+//! peer connection and nothing more. However, we still use the network graph because onion messages
+//! between channel peers are likely to be prioritized over those between non-channel peers.
```

Comment:
So while I think relying on the existence of channels between peers to flow onion messages align well the incentives between bandwidth consumed and onion forwards accepted, I don't know if this is a direction we would like for the long-term in term of privacy. Indeed, overlapping both offers flows and HTLC flows, if we re-use the same heuristics than for classic pathfinding we might have overlap and show more "data trails" to any payment observers. Especially onion messages routing might invalidate some privacy-preserving enhancing we have implemented with HTLCs such as https://github.com/lightningdevkit/rust-lightning/pull/1286. The lack of timing padding in onion routing might cancel the CLTV delta offset. Of course, this is a subject we can think more and improve with time. Just to be aware of about the privacy regression we might have.